### ISSUE TITLE
Correction in the code

### ISSUE THREAD

**Issue by vinodkkurmi** [Khurramjaved96/iCarl2.0#5]
hi @khurramjaved96  
There should be a small change in your code. In the trainer.py file, in the distillation loss section, the previous model giving features after passing through the softmax while the current model giving features by passing through log softmax. How we can compute the KL divergence between these two using the F.kl function?
Please check this. 
 Thanks

**Comment by kjaved0** [comment]
Hey, 
This is how the kl_div function is implemented in PyTorch.

The documentation of the function says:

"As with NLLLoss, the input given is expected to contain log-probabilities and is not restricted to a 2D Tensor. The targets are given as probabilities (i.e. without taking the logarithm)."

Source: https://pytorch.org/docs/master/nn.html?highlight=kldiv#torch.nn.KLDivLoss
